testing/integration/endpoint_security_test.go

```diff
@@ -493,7 +493,7 @@ func TestEndpointSecurityNonDefaultBasePath(t *testing.T) {
 	info := define.Require(t, define.Requirements{
 		Stack:   &define.Stack{},
 		Local:   false, // requires Agent installation
-		Isolate: false,
+		Isolate: true,
```

Comment:
Why all the changes to isolate? That means this will add 4 more machines to perform tests, why?

This seems unrelated to the change implemented in this PR. I didn't see anything in the PR description describing this change and its reason.

Reply:
That was me trying to get the integration tests to pass. Some exploratory testing because I just can't reproduce any of the CI failures when running the tests from my machine. I undid this change.

But I still have the changes from https://github.com/elastic/elastic-agent/pull/3786 to try getting CI happy. 

I'll request a re-review once it's the final code that can pass CI.